### PR TITLE
Feature/signup

### PR DESCRIPTION
**Pull request by Aurel-GitHub**
add register
add validator in register
add logout
update login with validators
sorry for the big bordel

### CODE REVIEW

File: src/pages/Profile.tsx [hunk 1]
```diff
@@ -184,25 +189,9 @@ const Profile = (): ReactElement => {
                   flexDirection="column"
                   placeholder="Email"
                   type="email"
-                  {...register('email')}
+                  {...register('email', { pattern: /^\S+@\S+\.\S+$/ })}
```
Comment (JIdayyy) [inline src/pages/Profile.tsx:192]: You can remove this regex and use YUP resolver

File: src/pages/Register.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable react/jsx-props-no-spreading */
+import { Button, Flex, FormControl, Input, Text } from '@chakra-ui/react'
+import { FieldValues, useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+import { useCookies } from 'react-cookie'
```
Comment (JIdayyy) [inline src/pages/Register.tsx:6]: woopsy ! This should be removed

File: src/pages/Profile.tsx (hunk 1 above)
Comment (Aurel-GitHub) [reply src/pages/Profile.tsx:192]: merde je l'ai laissé

File: src/pages/Register.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        <FormControl p={10} w={['90%', '80%', '60%', '50%']}>
+          <Input
+            variant="flushed"
+            placeholder="Firstname"
+            my={2}
+            type="text"
+            isInvalid={errors.first_name && true}
+            {...register('first_name', { required: true })}
+          />
+          <ErrorMessage
```
[the remark this answers is not in the document]
Comment (JIdayyy) [reply src/pages/Register.tsx:96]: And we should remove the required props here !

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -64,27 +84,44 @@ export default function Login(): JSX.Element {
           <Input
```
[the remark this answers is not in the document]
Comment (Aurel-GitHub) [reply src/pages/Login.tsx:84]: j'ai prévu de m'en occuper  sur cette carte: https://www.notion.so/144f35a324e44bf79392be1836e46d46?v=1c0a63341d6f4b3a98d40e500799935a&p=ddf675f2f6454c2ea32df1380a4e65b5

**Pull request merged**